sonar-db/src/main/java/org/sonar/db/issue/IssueFilterDao.java

```diff
@@ -72,27 +81,45 @@ public IssueFilterDto selectProvidedFilterByName(String name) {
     }
   }
 
+  /**
+   * @deprecated since 5.2 use {@link #selectSharedFilters(DbSession)}
+   */
+  @Deprecated
   public List<IssueFilterDto> selectSharedFilters() {
-    SqlSession session = mybatis.openSession(false);
+    DbSession session = mybatis.openSession(false);
     try {
-      return mapper(session).selectSharedFilters();
+      return selectSharedFilters(session);
     } finally {
       MyBatis.closeQuietly(session);
     }
   }
 
+  public List<IssueFilterDto> selectSharedFilters(DbSession session) {
+    return mapper(session).selectSharedFilters();
+  }
+
+  /**
+   * @deprecated since 5.2 use {@link #insert(DbSession, IssueFilterDto)}
+   */
+  @Deprecated
   public void insert(IssueFilterDto filter) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)